arangod/Transaction/Manager.cpp

```diff
@@ -499,7 +523,7 @@ void Manager::returnManagedTrx(TRI_voc_tid_t tid, AccessMode::Type mode) noexcep
   WRITE_LOCKER(writeLocker, _transactions[bucket]._lock);
 
   auto it = _transactions[bucket]._managed.find(tid);
-  if (it == _transactions[bucket]._managed.end()) {
+  if (it == _transactions[bucket]._managed.end() || !::authorized(it->second.user)) {
     LOG_TOPIC("1d5b0", WARN, Logger::TRANSACTIONS)
         << "managed transaction was not found";
     TRI_ASSERT(false);
```

Comment:
seems like the warning and the assertion are not meant to be used

Reply:
I believe you are correct, and any codepath that leads here would already go through some sort of user-checking, so we can safely ignore it here.